extensions/datasource/deployment-spi/src/main/java/io/quarkus/datasource/deployment/spi/DatabaseDefaultSetupConfig.java

```diff
@@ -0,0 +1,11 @@
+package io.quarkus.datasource.deployment.spi;
+
+public interface DatabaseDefaultSetupConfig {
+
+    String DEFAULT_DATABASE_USERNAME = "quarkus";
+
+    // mssql container enforces a 'strong' password with min 8 chars, upper/lowercase, number or special char
+    // db2 derives the instance name from this. Enforces max 8 chars, only lowercase, picky on special chars
```

Comment:
There's a chance this comment is incorrect and the problem is due to the big mix up between usernames and passwords.